jenkins/promotion/release-promotion.jenkinsfile

```diff
@@ -0,0 +1,433 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+lib = library(identifier: 'jenkins@5.11.1', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/opensearch-project/opensearch-build-libraries.git',
+]))
+
+pipeline {
+    options {
+        timeout(time: 4, unit: 'HOURS')
+    }
+    agent none
+    environment {
+        AGENT_LINUX_X64 = 'Jenkins-Agent-AL2023-X64-C54xlarge-Docker-Host'
+    }
+    parameters {
+        string(
+            name: 'RELEASE_VERSION',
+            description: 'Release version',
+            trim: true
+        )
+        string(
+            name: 'OPENSEARCH_RC',
+            description: 'OpenSearch Release Candidate',
+            trim: true
+        )
+        string(
+            name: 'OPENSEARCH_DASHBOARDS_RC',
+            description: 'OpenSearch Dashboards Release Candidate',
+            trim: true
+        )
+    }    
+    stages {
+        stage('Parameters Check') {
+            steps {
+                script {
+                    currentBuild.description = """RELEASE: ${RELEASE_VERSION}<br>OS_RC: ${OPENSEARCH_RC}<br>OSD_RC: ${OPENSEARCH_DASHBOARDS_RC}"""
+                    if(RELEASE_VERSION.isEmpty() || OPENSEARCH_RC.isEmpty() || OPENSEARCH_DASHBOARDS_RC.isEmpty()) {
+                                currentBuild.result = 'ABORTED'
+                                error('Make sure all the parameters are passed in.')
+                    }
+                }
+            }
+        }
+        stage('OpenSearch Apt promotion') {
+            agent {
+                docker {
+                    label AGENT_LINUX_X64
+                    image 'docker/library/alpine:3'
+                    registryUrl 'https://public.ecr.aws/'
+                    alwaysPull true
+                }
+            }
+            steps {
+                echo 'Triggering distribution-promote-repos for OpenSearch apt'
+                build job: 'distribution-promote-repos', wait: true, parameters: [choice(name: 'DISTRIBUTION_JOB_NAME', value: 'distribution-build-opensearch'), 
+                                                                                    choice(name: 'DISTRIBUTION_REPO_TYPE', value: 'apt'),
+                                                                                    string(name: 'DISTRIBUTION_BUILD_NUMBER', value: params.OPENSEARCH_RC),
+                                                                                    string(name: 'INPUT_MANIFEST', value: "${params.RELEASE_VERSION}/opensearch-${params.RELEASE_VERSION}.yml"),
```

Comment:
Should we take the input manifest as paramter instead of dynamically forming the string? Maybe in next iteration? As of now cannot think of scenario we would name input manifest something else.

Reply:
Sure, we can keep an eye on this and update in next iteration, just want to avoid a manual input for a release manager.